backend/copr_backend/background_worker_build.py

```diff
@@ -436,6 +455,15 @@ def _alloc_ssh_connection(self):
             log=self.log,
         )
 
+    def _discard_running_worker(self):
+        """
+        This is "canceling" callback to CancellableThreadTask, so please never
+        raise any exception.  The worst case scenario is that nothing is
+        canceled.
+        """
+        if self.canceled:
+            self._cancel_running_worker()
```

Comment:
small nit, we could make the patch smaller by moving this `self.canceled` condition into the `_cancel_running_worker()` task (perhaps renaming the method, and keeping the comment).